{
  Exception testException=getException();
  Throwable t=ExceptionHelper.getRootException(testException);
  assertNotNull(t);
  assertThat(t.getMessage(),is("blah"));
  assertThat(t.getCause(),nullValue());
  t=ExceptionHelper.getRootMuleException(testException);
  assertThat(t.getMessage(),is("bar"));
  assertThat(t.getCause(),not(nullValue()));
  List<Throwable> l=ExceptionHelper.getExceptionsAsList(testException);
  assertThat(l,hasSize(3));
  Map<String,String> info=ExceptionHelper.getExceptionInfo(testException);
  assertThat(info.entrySet(),hasSize(2));
  assertThat(info,hasEntry("info_1","Imma in!"));
  assertThat(info,hasEntry("info_2","Imma out!"));
}
